### PR TITLE
Emitter tests

### PR DESCRIPTION
**Pull request by alfredclwong**
## Changes

- Bug fix: calling `AngularPower.Cosine()` caused an `UndefVarError`
- simplified secondary constructors in Directions.jl
- code style improvements
- `generateray` -> `generate` (pls review: 88856f5)
- test coverage for Emitters module

### CODE REVIEW

File: src/Optical/Emitters/AngularPower.jl
Comment (alfredclwong) [inline src/Optical/Emitters/AngularPower.jl:36]: calling `Cosine()` didn't provide any information for type inference - we need to give a default type

File: src/Optical/Emitters/Directions.jl
Comment (alfredclwong) [inline src/Optical/Emitters/Directions.jl:77]: this simplifies the constructor a bit without changing functionality

File: src/Optical/Emitters/Origins.jl
Comment (alfredclwong) [inline src/Optical/Emitters/Origins.jl:17]: i think this is the right correction - @galran can you confirm?

File: src/Optical/Emitters/AngularPower.jl
Comment (BrianGun) [reply src/Optical/Emitters/AngularPower.jl:36]: @alfredclwong this is correct but could be confusing to new users. Maybe break into two constructors?

function Cosine(cosine_exp::T) where {T<:Real} = new{T}(cosine_exp)
function Cosine() = new{T}(one(Float64))

It's a small point, if you want to leave it then I'll approve anyway.

**Comment by BrianGun** [comment]
are we just waiting for @galran to approve the getindex change? Once that is approved can we approve the PR?

**Comment by alfredclwong** [comment]
> are we just waiting for @galran to approve the getindex change? Once that is approved can we approve the PR?

Confirmation would be nice - I'll ask today at the meeting. It's most likely correct anyway, but good to be safe.

**Comment by galran** [comment]
looks ok to me.
i'm ashamed to say it but i don't remember how to "officially" approve it :)

**Comment by alfredclwong** [comment]
> looks ok to me.
> i'm ashamed to say it but i don't remember how to "officially" approve it :)

It's pretty obscure - you have to go to "Files changed", click "Review changes", select "Approve", and then "Submit review". You can also use the other options to create review comments without approving the PR.

**Review by friggog — APPROVED** [review]

**Pull request merged**
